opencti-platform/opencti-front/src/private/components/common/location/LocationCreation.tsx

```diff
@@ -0,0 +1,276 @@
+  }
+`;
+
+interface LocationAddInput {
+  name: string
+  description: string
+  type: string
+}
+
+interface LocationCreationFormProps {
+  updater: (store: RecordSourceSelectorProxy) => void;
+  onReset?: () => void;
+  display?: boolean
+  contextual?: boolean
+  onCompleted?: () => void;
+  inputValue?: string;
+  creationCallback?: (data:LocationCreationMutation$data) => void;
+}
+
+const locationValidation = (t) => Yup.object().shape({
+  name: Yup.string().required(t('This field is required')),
+  description: Yup.string().nullable(),
+  type: Yup.string().required(t('This field is required')),
+});
+
+const LocationCreationForm: FunctionComponent<LocationCreationFormProps> = ({
+  inputValue,
+  onlyAuthors,
+  onReset,
+  onCompleted,
+  contextual,
+  creationCallback,
+  updater,
+}) => {
+  const classes = useStyles();
+  const { t } = useFormatter();
+
+  const [commit] = useMutation<LocationCreationMutation>(locationMutation);
+
+  const onSubmit: FormikConfig<LocationAddInput>['onSubmit'] = (
+    values,
+    {
+      setSubmitting,
+      resetForm,
+      setErrors,
+    },
+  ) => {
+    const input: LocationCreationMutation$variables['input'] = {
+      name: values.name,
+      description: values.description,
+      type: values.type,
+    };
+    commit({
+      variables: {
+        input,
+      },
+      updater: (store) => updater(store),
+      onError: (error: Error) => {
+        handleErrorInForm(error, setErrors);
+        setSubmitting(false);
+      },
+      onCompleted: (response) => {
+        setSubmitting(false);
+        resetForm();
+        if (contextual && creationCallback) {
+          creationCallback(response);
+        }
+        if (onCompleted) {
+          onCompleted();
+        }
+      },
+    });
+  };
+
+  return (
+    <Formik
+      enableReinitialize={true}
+      initialValues={{
+        name: inputValue,
+        description: '',
+        type: '',
+      }}
+      validationSchema={locationValidation(t)}
+      onSubmit={onSubmit}
+      onReset={onReset}
+    >
+      {({
+        submitForm,
+        handleReset,
+        isSubmitting,
+      }) => (
+        <Form style={{ margin: '20px 0 20px 0' }}>
+          <Field
+            component={TextField}
+            variant="standard"
+            name="name"
+            label={t('Name')}
+            fullWidth={true}
+            detectDuplicate={['Organization', 'Individual']}
+          />
+          <Field
+            component={MarkdownField}
+            name="description"
+            label={t('Description')}
+            fullWidth={true}
+            multiline={true}
+            rows="4"
+            style={{ marginTop: 20 }}
+          />
+          <Field
+            component={SelectField}
+            variant="standard"
+            name="type"
+            label={t('Entity type')}
+            fullWidth={true}
+            containerstyle={fieldSpacingContainerStyle}
+          >
+            {!onlyAuthors && (
```

Comment:
maybe you should do a .map on the LOCATION_ENTITIES list (instead of hardcoding the different location types)

Reply:
Actually, I'm not sure if I can do that. The MenuItem component of Mui seems to take only one value not multiple. But I will definitely check on the possibility to do it!

Reply:
you still pass 1 value to MenuItem ... you do something like that:
{!onlyAuthors && LOCATION_ENTITIES.map((location) => (
    <MenuItem value=location>{t(location)}</MenuItem>
)}

Reply:
I did and it wasn't selecting a value that's why I was saying that I wasn't sure of the possibility, but I will try again differently ! :)